plugins/modules/ec2_asg_scheduled_action.py

```diff
@@ -0,0 +1,329 @@
+        return False
+
+    if module.check_mode:
+        return True
+
+    params = dict(
+        AutoScalingGroupName=module.params.get('autoscaling_group_name'),
+        ScheduledActionName=module.params.get('scheduled_action_name')
+    )
+
+    try:
+        client.delete_scheduled_action(aws_retry=True, **params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    return True
+
+
+def get_scheduled_actions():
+    params = dict(
+        AutoScalingGroupName=module.params.get('autoscaling_group_name'),
+        ScheduledActionNames=[module.params.get('scheduled_action_name')]
+    )
+
+    try:
+        actions = client.describe_scheduled_actions(aws_retry=True, **params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json_aws(e)
+
+    current_actions = actions.get("ScheduledUpdateGroupActions")
+
+    return current_actions
+
+
+def put_scheduled_update_group_action(current_actions):
+    changed = False
+    changes = dict()
+    params = format_request()
+
+    if len(current_actions) < 1:
+        changed = True
+    else:
+        for k, v in params.items():
+            # To correctly detect changes convert the start_time & end_time to datetime object
+            if k == "StartTime":
+                v = timedate_parse(v)
+            elif k == "EndTime":
+                v = timedate_parse(v)
+
+            if current_actions[0].get(k) != v:
+                changes[k] = v
+
+        if changes:
+            changed = True
+
+    if module.check_mode:
+        return changed
+
+    try:
+        client.put_scheduled_update_group_action(aws_retry=True, **params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json_aws(e)
+
+    return changed
+
+
+def main():
+    global module
+    global client
+
+    argument_spec = dict(
+        autoscaling_group_name=dict(required=True, type='str'),
+        scheduled_action_name=dict(required=True, type='str'),
+        start_time=dict(default=None, type='str'),
+        end_time=dict(default=None, type='str'),
+        time_zone=dict(default=None, type='str'),
+        recurrence=dict(type='str'),
+        min_size=dict(default=None, type='int'),
+        max_size=dict(default=None, type='int'),
+        desired_capacity=dict(default=None, type='int'),
+        state=dict(default='present', choices=['present', 'absent'])
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        required_if=[['state', 'present', ['recurrence']]],
+        supports_check_mode=True
+    )
+
+    if not HAS_DATEUTIL:
+        module.fail_json(msg='dateutil is required for this module')
+
+    client = module.client('autoscaling', retry_decorator=AWSRetry.jittered_backoff())
+    current_actions = get_scheduled_actions()
+    state = module.params.get('state')
+    results = dict()
+
+    if state == 'present':
+        changed = put_scheduled_update_group_action(current_actions)
```

Comment:
This is not covered by `check_mode`.

Reply:
@markuman so it calls the function and has a check for check mode here:
https://github.com/ansible-collections/community.aws/pull/779/files#diff-9af6ca29f1ac5f35e2679d6d8eadc230ac054671a271f2d50d8ce9e0898a3866R265